testing/test_recwarn.py

```diff
@@ -545,6 +545,38 @@ def test_it():
         result.assert_outcomes()
 
 
+def test_re_emit_uses_correct_module(pytester: Pytester) -> None:
+    warning_module_code = """
+import warnings
+
+def trigger_warning(msg):
+    warnings.warn(msg, UserWarning)
+"""
+    pytester.makepyfile(module_a=warning_module_code)
+    pytester.makepyfile(module_b=warning_module_code)
+
+    test_code = """
+        import pytest
+    import warnings
+    from module_a import trigger_warning as trigger_warning_a
+    from module_b import trigger_warning as trigger_warning_b
+
+    def test_ignore_warning_from_module_a():
+        with pytest.raises(pytest.fail.Exception, match="DID NOT WARN"):
+            with pytest.warns(UserWarning, match="module A.") as outer:
```

Comment:
The test is not really veryfing the new `module` attribute... if I go in the code and set `module = None` in `warnings.warn_explicit`, the test still passes, which makes it unsuitable for a regression test... can you improve the test to check the actual `module` attribute of the warnings?